androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/LauncherActivity.java

```diff
@@ -347,18 +375,27 @@ public void onEnterAnimationComplete() {
      * Override this for special handling (such as ignoring or sanitising data from the Intent).
      */
     protected Uri getLaunchingUrl() {
-        Uri uri = getIntent().getData();
-        if (uri != null) {
-            Log.d(TAG, "Using URL from Intent (" + uri + ").");
-            return uri;
-        }
-
-        if (mMetadata.defaultUrl != null) {
-            Log.d(TAG, "Using URL from Manifest (" + mMetadata.defaultUrl + ").");
-            return Uri.parse(mMetadata.defaultUrl);
+        Uri baseUrl = Uri.parse(mMetadata.defaultUrl);
+
+        Uri intentUrl = getIntent().getData();
+
+        if (intentUrl != null) {
+            Map<String, Uri> protocolHandlers = getProtocolHandlers();
+            String scheme = intentUrl.getScheme();
+            if (protocolHandlers.containsKey(scheme)) {
+                String format = Objects.requireNonNull(protocolHandlers.get(scheme)).toString();
+                String target = intentUrl.toString().replace(scheme + "://", "");
```

Comment:
In accordance to my above comment, I don't _think_ you want to be stripping the scheme here.